docs/data_structures/gen_specs.rst

```diff
@@ -1,59 +1,80 @@
 .. _datastruct-gen-specs:
 
-gen_specs
-=========
-
-Generation function specifications to be set in user calling script and passed
-to main ``libE()`` routine::
-
-    gen_specs: [dict]:
-
-        'gen_f' [func]:
-            generates inputs to sim_f
-        'in' [list]:
-            field names (as strings) that will be given to gen_f function when
-            it is called (if not present, then no fields will be passed).
-        'persis_in' [list]:
-            field names (as strings) that will be given back to a persistent gen_f
-        'out' [list of tuples (field name, data type, [size])]:
-            gen_f outputs that will be stored in the libEnsemble history
-        'user' [dict]:
-            Data structure to contain problem specific constants and/or input data
+Generator Specs
+===============
+
+Used to specify the generator function, its inputs and outputs, and user data. Can be constructed and passed to libEnsemble
+as a Python class (with automatic input validation) or a simple dictionary.
```

Comment:
Is input validation used with the dictionary also? This wording to me suggests only when provided as a class.